massa-bootstrap/src/messages.rs

```diff
@@ -177,24 +195,36 @@ impl Serializer<BootstrapServerMessage> for BootstrapServerMessageSerializer {
                     .serialize(graph, buffer)?;
             }
             BootstrapServerMessage::FinalStatePart {
-                ledger_data,
+                slot,
+                ledger_part,
                 async_pool_part,
                 pos_cycle_part,
                 pos_credits_part,
                 exec_ops_part,
-                slot,
                 final_state_changes,
             } => {
+                // message type
                 self.u32_serializer
                     .serialize(&u32::from(MessageServerTypeId::FinalStatePart), buffer)?;
-                self.vec_u8_serializer.serialize(ledger_data, buffer)?;
-                self.vec_u8_serializer.serialize(async_pool_part, buffer)?;
-                self.vec_u8_serializer.serialize(pos_cycle_part, buffer)?;
-                self.vec_u8_serializer.serialize(pos_credits_part, buffer)?;
-                self.vec_u8_serializer.serialize(exec_ops_part, buffer)?;
+                // slot
                 self.slot_serializer.serialize(slot, buffer)?;
-                self.u32_serializer
-                    .serialize(&(final_state_changes.len() as u32), buffer)?;
+                // ledger
+                self.vec_u8_serializer.serialize(ledger_part, buffer)?;
+                // async pool
+                self.async_pool_serializer
+                    .serialize(async_pool_part, buffer)?;
+                // pos cycle info
+                self.opt_pos_cycle_serializer
+                    .serialize(pos_cycle_part, buffer)?;
+                // pos deferred credits
+                self.pos_credits_serializer
+                    .serialize(pos_credits_part, buffer)?;
+                // executed operations
+                self.exec_ops_serializer.serialize(exec_ops_part, buffer)?;
+                // changes length
+                self.u64_serializer
```

Comment:
The len here is now u64 instead of u32 but hasn't been changed in the deserializer